llumnix/global_scheduler/migration_scheduler.py

```diff
@@ -66,12 +113,15 @@ def remove_instance(self, instance_id: str) -> None:
     def _sort_instance_infos(self,
                              descending: bool = True) -> None:
         instance_infos: List[InstanceInfo] = list(self.instance_info.values())
+        filter_instance_infos  = {instance_type: set() for instance_type in InstanceType}
         key_attr = 'instance_load_migrate'
-        self.sorted_instance_infos = sorted(
-            instance_infos,
-            key=lambda instance_info: getattr(instance_info, key_attr),
-            reverse=descending
-        )
+        for inst_type in InstanceType:
+            filter_instance_infos[inst_type] = [info for info in instance_infos if info.instance_id in self.instance_id_type_set[inst_type]]
```

Comment:
> Why not mark instance_type into instance_info, sort once, and filter different lists? If our sorted key remains unchanged, sort multiple times is unnecessary.

It might be not suitable to mark instance_type(like prefill/decode) into instance_info. The instance type should only be perceived and managed by the scheduler. 
I've already made changes according to your second comment. It also can reduce the sorting complexity.

Reply:
My point was that we can directly mark instance_type in migration scheduler or global_scheduler. Just like we mark instance_load_migrate and instance_load_migrate in global_scheduler. I don't know whether it is still necessary to do so after your modification now, you can refer to it.

Reply:
lgtm